crates/red_knot_python_semantic/src/types/call.rs

```diff
@@ -249,6 +258,54 @@ impl<'db> CallOutcome<'db> {
                     }),
                 }
             }
+            CallOutcome::StaticAssertionError {
+                parameter_ty: Type::BooleanLiteral(false),
+                truthiness: _,
+            } => {
+                context.report_lint(
+                    &STATIC_ASSERT_ERROR,
+                    node,
+                    format_args!("Static assertion error: argument evaluates to `False`"),
+                );
+
+                Ok(Type::Unknown)
+            }
```

Comment:
Is there value in separating this case from the one immediately below? I found this surprising when I saw it in the tests.

Reply:
I wanted a slightly easier to understand message for the (presumably most common) case where your assertion condition actually evaluates to `False`. Like in a comparison that's simply not true.

And I didn't want to re-use this message ("evaluates to `False`") for cases like `static_assert(<expression of type None>)` because it could be confusing, and because I wanted to add more context (show the actual type in the message), which seems not very helpful for cases where the type is `Literal[False]`.